### PR TITLE
Use hipblasltDatatype_t instead of hipblasDatatype_t for softmax API

### PR DESCRIPTION
**Pull request by Serge45**
## Brief ##
Change the datatype of interface for softmax API from `hipblasDatatype_t` to `hipblasltDatatype_t`.

### CODE REVIEW

**Review by TonyYHsieh — APPROVED** [review]
LGTM

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 1]
```diff
@@ -1146,23 +1117,34 @@ namespace
...
-    case HIPBLAS_R_32F:
-        return Tensile::DataType::Float;
-    case HIPBLAS_R_16B:
-        return Tensile::DataType::BFloat16;
-    default:
-        assert(!"hipblasDatatype_to_tensile_type: non-supported type");
-        return Tensile::DataType::None;
+ * Map a hipblaslt data type to a corresponding Tensile type *
+ ******************************************************/
+    Tensile::DataType hipblasltDatatype_to_tensile_type(hipblasltDatatype_t type)
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp:1122]: why did you remove the inline?

Comment (Serge45) [reply library/src/amd_detail/rocblaslt/src/tensile_host.cpp]: Corrected, lost the specifier during editing.

**Review by jichangjichang — APPROVED** [review]
lgtm, merge it after CI pass
